modules/_node-scripts/bundle/sass/processSassFiles.mjs

```diff
@@ -0,0 +1,203 @@
+ */
+
+import fs from 'fs/promises';
+import {r2} from 'liferay-theme-tasks';
+import path from 'path';
+import resolve from 'resolve';
+
+import {BUILD_RESOURCES_PATH, BUILD_SASS_CACHE_PATH, SRC_PATH} from '../../util/constants.mjs';
+import expandGlobs from '../../util/expandGlobs.mjs';
+import projectScopeRequire from '../../util/projectScopeRequire.mjs';
+import runSass from './runSass.mjs';
+
+const CSS_IMPORT_REGEX = /@import\s+url\s*\(\s*['"]?(.+?\.css)/g;
+const SASS_EXCLUDE = [
+	'**/_diffs/**',
+	'**/_sass_cache_*/**',
+	'**/_styled/**',
+	'**/_unstyled/**',
+	'**/.sass-cache*/**',
+	'**/build/**',
+	'**/classes/**',
+	'**/css/aui/**',
+	'**/css/clay/**',
+	'**/tmp/**',
+];
+
+export default async function processSassFiles() {
+	const {entryFiles, partials} = await collectSassFiles();
+
+	if (!entryFiles.length) {
+		return;
+	}
+
+	if (!await isAnyFileModified([...entryFiles, ...partials], SRC_PATH, BUILD_RESOURCES_PATH)) {
+		console.log('⚠️ No .scss files were modified: skipping Sass compilation');
+		return;
+	}
+
+	await Promise.all([
+		...entryFiles.map(copySassFile),
+		...partials.map(copySassFile),
+	]);
+
+	const includePaths = getIncludePaths();
+
+	const timestamp = Date.now();
+
+	await Promise.all(
+		entryFiles.map(async (entryFile) => {
+			const start = performance.now();
+
+			await processSassFile(entryFile, includePaths, timestamp);
+
+			const lapse = performance.now() - start;
+
+			console.log(
+				`⌛ Sass for ${entryFile} took: ${(lapse / 1000).toFixed(3)} s`
+			);
+		})
+	);
+}
+
+/**
+ * Append `t=<timestamp>` query string parameters to all `@import`ed files inside a CSS file.
+ *
+ * This is done for historical reasons even though the technique is completely flawed and creates a
+ * lot of problems with HTTP caching.
+ */
+function appendTimestamps(css, timestamp) {
+	const matches = css.match(CSS_IMPORT_REGEX) || [];
+
+	for (const match of matches) {
+		css = css.replace(match, `${match}?t=${timestamp}`);
+	}
+
+	return css;
+}
+
+async function collectSassFiles() {
+	const files = await expandGlobs(['**/*.scss'], SASS_EXCLUDE, {baseDir: SRC_PATH});
+
+	return files.reduce(
+		(acc, file) => {
+			if (path.basename(file).match(/^_.+\.scss$/)) {
+				acc.partials.push(file);
+			}
+			else {
+				acc.entryFiles.push(file);
+			}
+
+			return acc;
+		},
+		{entryFiles: [], partials: []}
+	);
+}
+
+async function copySassFile(filePath) {
+	const destFilePath = path.join(BUILD_RESOURCES_PATH, path.relative(SRC_PATH, filePath));
+
+	await fs.mkdir(path.dirname(destFilePath), {recursive: true});
+
+	await fs.copyFile(filePath, destFilePath);
+
+	const stats = await fs.stat(filePath);
+
+	await fs.utimes(destFilePath, stats.atime, stats.mtime);
+}
+
+function getIncludePaths() {
+	const includePaths = [
+		path.dirname(resolve.sync('bourbon', {basedir: '.'})),
+		path.dirname(resolve.sync('liferay-frontend-common-css', {basedir: '.'}))
```

Comment:
any idea if bourbon still necessary? Seems ancient

Reply:
no idea, just copied it over from npm-scripts